src/smt/set_defaults.cpp

```diff
@@ -350,6 +350,18 @@ void setDefaults(SmtEngine& smte, LogicInfo& logic)
     }
   }
 
+
+  if (options::solveBVAsInt() > 0)
+  {
+    /**
+     * Operations on 1 bits are better handled as Boolean operations
+     * than as integer operations.
+     * Therefore, we enable bv-to-bool, which runs before
+     * the translation to integers.
+     */
+    options::bitvectorToBool.set(true);
+  }
+
```

Comment:
Is pulling this out of the unsat-core/proof conditional intentional?

Reply:
yes, this was there by mistake. bv-to-bool should be turned on regardless of proofs/unsat cores.